proposals/0016-c-interface-compiler-library.md

```diff
@@ -0,0 +1,84 @@
+<!-- {% raw %} -->
+
+# C interface for HLSL compiler as a library
+
+* Proposal: [0016](0016-c-interface-compiler-library.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+  [#63631](https://github.com/llvm/llvm-project/issues/63631)
+
+## Introduction
+
+An effort is underway to modernize compilation of HLSL based shaders. This is
+commonly referred to as the upstream effort. This effort is a long term play
+that positions HLSL shader compilation in a place that provides the most
+stability and maintainability for the future by giving as many options to
+DirectX while the GPU/NPU landscape evolves.
+
+There are multiple ways to compile a shader today.
+* dxc.exe - Executable that takes command line options and outputs bytecode
+* IDxc** interfaces - COM-based library that is consumed by toolchains to
+compile shaders.
+
+## Motivation
+
+One of the HLSL compiler's primary usage modes is as a library. The upstream
+compiler will also need to expose a library for toolchains to compile a shader.
+
+## Proposed solution
+
+The existing HLSL compiler's library support on Windows is COM based.
+This is a non-starter for the upstream effort.  COM doesn't fit any of the
```

Comment:
"upstream effort"

I wonder at what point we start referring to this just as "clang" or something like that, rather than "upstream".  Upstream tends to frame everything from the perspective of where we were, rather than of where we are / want to be.

Reply:
+1 this. I've been trying to socialize calling the new HLSL compiler "Clang" too. I think that really helps clarify our commitment to the open source project and community rather than drawing an arbitrary distinction.